src/main/java/com/algorand/algosdk/v2/client/common/Client.java

```diff
@@ -30,7 +30,21 @@ public String getHost() {
     }
 
     public static HttpUrl getHttpUrl(QueryData qData, int port, String host) {
-        HttpUrl.Builder httpUrlBuilder = (new HttpUrl.Builder()).scheme("http").port(port).host(host);
+        HttpUrl.Builder httpUrlBuilder = null;
+        HttpUrl parsedHttpUrl = HttpUrl.parse(host);
+
+        if (parsedHttpUrl != null) {
+            httpUrlBuilder = parsedHttpUrl.newBuilder();
+
+            // If the default port is being used, override it with the port provided to our constructor.
+            if (HttpUrl.defaultPort(parsedHttpUrl.scheme()) == parsedHttpUrl.port()) {
+                httpUrlBuilder.port(port);
+            }
+        } else {
+            httpUrlBuilder = new HttpUrl.Builder();
+            httpUrlBuilder.scheme("http").port(port).host(host);
+        }
```

Comment:
I wouldn't correct the user's behavior. If the user provided 2 different ports, throw an error. It's ok to request the port in a separate field.